cmd/neofs-node/reputation.go

```diff
@@ -309,15 +309,21 @@ func initReputationService(c *cfg) {
 		LocalTrustTarget: router,
 	})
 
-	addNewEpochNotificationHandler(c, func(ev event.Event) {
-		var reportPrm trustcontroller.ReportPrm
-
-		// report collected values from previous epoch
-		reportPrm.SetEpoch(ev.(netmap.NewEpoch).EpochNumber() - 1)
-
-		// TODO: implement and use worker pool [neofs-node#440]
-		go c.cfgReputation.localTrustCtrl.Report(reportPrm)
-	})
+	addNewEpochNotificationHandler(
```

Comment:
1. What about others events? I think some of them (if not all) will benefit from async execution. Consider `cmd/neofs-node/netmap.go:77` and `cmd/neofs-node/netmap.go:89` for example.

2. If some handlers are designed by a developer to be async and some are not, then I would expect to have two functions: `addNewEpochNotificationHandler` and `addNewEpochAsyncNotificationHandler` (or at least bool argument for async). This way you won't need to copy-paste `event.WorkerPoolHandler` wrapper in all handler registrations.

```go
// netmap.go
func addNewEpochNotificationHandler(c *cfg, h event.Handler) {
	addNetmapNotificationHandler(c, newEpochNotification, h)
}

func addNewEpochAsyncNotificationHandler(c *cfg, h event.Handler) {
	addNetmapNotificationHandler(c, newEpochNotification,
		event.WorkerPoolHandler(
			c.cfgNetmap.workerPool,
			h,
			c.log,
		))
}
...
// sync handler example
addNewEpochNotificationHandler(c, func(ev event.Event) {
	c.cfgNetmap.state.setCurrentEpoch(ev.(netmapEvent.NewEpoch).EpochNumber())
})

if c.cfgNetmap.reBootstrapEnabled {
	// async handler example
	addNewEpochAsyncNotificationHandler(c, func(ev event.Event) {
		n := ev.(netmapEvent.NewEpoch).EpochNumber()
		...
		}
	})
}

```

Reply:
1. did not know, what could be done async. and are there any handlers that MUST be sync or not, so decided to keep them as they are now. can change handlers that you mentioned or try to make all of them(container, object and so on) async.
2. agree, added such func to netmap and similar func to container in `main` pkg.

Reply:
>  can change handlers that you mentioned or try to make all of them(container, object and so on) async.

I suggest you to make async handlers for all operations that use network communication, as it stated in the original issue. I found those:
- [ ] https://github.com/carpawell/neofs-node/blob/b0d023d522ffbb9d9ab262d813c7615d909b6b18/cmd/neofs-node/netmap.go#L77 (invokes `AddPeer` method at network map contract)
- [ ] https://github.com/carpawell/neofs-node/blob/b0d023d522ffbb9d9ab262d813c7615d909b6b18/cmd/neofs-node/netmap.go#L89 (`netmapLocalNodeState()` fetches network map from the contract)
- [ ] https://github.com/carpawell/neofs-node/blob/b0d023d522ffbb9d9ab262d813c7615d909b6b18/cmd/neofs-node/container.go#L109 (announce local values to other storage nodes)
- [ ] https://github.com/carpawell/neofs-node/blob/b0d023d522ffbb9d9ab262d813c7615d909b6b18/cmd/neofs-node/container.go#L116 (sends estimate result to the contract)

Reply:
added